packages/grid/x-data-grid-premium/src/tests/clipboard.DataGridPremium.test.tsx

```diff
@@ -250,6 +250,34 @@ describe('<DataGridPremium /> - Clipboard', () => {
         expect(getCell(1, 2)).to.have.text(secondColumnValuesBeforePaste[1]);
         expect(getCell(2, 2)).to.have.text(secondColumnValuesBeforePaste[2]);
       });
+
+      it('should paste all values when they exceed `pageSize` and pagination disabled but fit into visible row area', async () => {
+        render(
+          <Test
+            rowLength={8}
+            colLength={4}
+            paginationModel={{ page: 1, pageSize: 5 }}
+            pagination={false}
+          />,
+        );
+
+        const cell = getCell(1, 1);
+        cell.focus();
+        userEvent.mousePress(cell);
+
+        const clipboardData = Array.from(new Array(7))
+          .map((_, idx) => [`p${idx + 1}1`, `p${idx + 1}2`, `p${idx + 1}3`].join('\t'))
+          .join('\n');
```

Comment:
To make it more explicit
```suggestion
        const clipboardData = [
          ['p11', 'p12', 'p13'],
          ['p21', 'p22', 'p23'],
          ['p31', 'p32', 'p33'],
          ['p41', 'p42', 'p43'],
          ['p51', 'p52', 'p53'],
          ['p61', 'p62', 'p63'],
          ['p71', 'p72', 'p73'],
        ]
          .map((row) => row.join('\t'))
          .join('\n');
```